### PR TITLE
Moved the NpmPackage dependency for structure-map based resource extraction.

### LINKED ISSUE #808: Remove NpmPackage dependency from datacapture library.
**Describe the Issue**
`NpmPackage` along with `packages.fhir.org-hl7.fhir.r4.core-4.0.1.tgz` are currently used in datacapture library for Structure-Map based extraction and it bloats the library because of its large size.
 
As a solution to this we would like to:
1. Either remove `packages.fhir.org-hl7.fhir.r4.core-4.0.1.tgz` dependency altogether from the datacapture library.
2. Or Move the Structure-Map based extraction into a separate module so that only clients who use  Structure-Map extraction have to include it.

**Would you like to work on the issue?**

### PR DESCRIPTION
**Pull request by aditya-07**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #808 

**Description**
Moved the NpmPackage dependency for structure-map based resource extraction action down from datacapture library to the client apps. Now the onus is on the client apps to load the NpmPackage with all the resources that maybe required by the structure-map.

This has reduced the size of our Reference App from **48.8 MB** to **35.6 MB**, `packages.fhir.org-hl7.fhir.r4.core-4.0.1.tgz` being **12.8MB**.
**Alternative(s) considered**
No

**Type**
Choose one:  Code health

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

**Comment by jingtang10** [comment]
> btw - have you noticed the size decrease before vs after this PR :) can you share the difference?

it's already in the pr description. thanks for pointing out.

File: datacapture/src/test/java/com/google/android/fhir/datacapture/mapping/ResourceMapperTest.kt [hunk 1]
```diff
@@ -1495,12 +1478,9 @@ class ResourceMapperTest {
...
     runBlocking {
       bundle =
-        ResourceMapper.extract(
-          uriTestQuestionnaire,
-          uriTestQuestionnaireResponse,
-          { map },
-          ApplicationProvider.getApplicationContext()
-        )
+        ResourceMapper.extract(uriTestQuestionnaire, uriTestQuestionnaireResponse) { _, worker ->
+          StructureMapUtilities(worker).parse(mapping, "FHIRMapperTutorial")
```
[the remark this answers is not in the document]
Comment (aditya-07) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/mapping/ResourceMapperTest.kt]: Done

File: datacapture/src/main/java/com/google/android/fhir/datacapture/utilities/SimpleWorkerContextProvider.kt [hunk 2]
```diff
@@ -37,13 +37,13 @@ object SimpleWorkerContextProvider {
    * The whole process can take 1-3 minutes on a clean installation, otherwise, it should take 20
    * seconds to 1 minute .
    */
-  suspend fun loadSimpleWorkerContext(context: Context): SimpleWorkerContext {
+  fun loadSimpleWorkerContext(): SimpleWorkerContext {
```
[the remark this answers is not in the document]
Comment (aditya-07) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/utilities/SimpleWorkerContextProvider.kt]: SimpleWorkerContext is now instantiated lazily in DataCaptureConfig itself.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt [hunk 3]
```diff
@@ -146,13 +148,14 @@ object ResourceMapper {
...
   ): Bundle {
-    if (structureMapProvider == null || context == null) return Bundle()
-    val structureMap = structureMapProvider(questionnaire.targetStructureMap!!) ?: return Bundle()
-    val simpleWorkerContext = SimpleWorkerContextProvider.loadSimpleWorkerContext(context)
+    if (structureMapProvider == null) return Bundle()
+    val simpleWorkerContext = SimpleWorkerContextProvider.loadSimpleWorkerContext()
     simpleWorkerContext.setExpansionProfile(Parameters())
+    val structureMap =
+      structureMapProvider(questionnaire.targetStructureMap!!, simpleWorkerContext)
+        ?: return Bundle()
```
[the remark this answers is not in the document]
Comment (aditya-07) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt]: Done

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 4]
```diff
@@ -103,9 +103,6 @@ object Dependencies {
   }
 
   const val androidJunitRunner = "androidx.test.runner.AndroidJUnitRunner"
-  const val apacheCommonsCompress =
-    "org.apache.commons:commons-compress:${Versions.apacheCommonsCompress}"
-  const val apacheCommonsIo = "commons-io:commons-io:${Versions.apacheCommonsIo}"
```
[the remark this answers is not in the document]
Comment (aditya-07) [reply buildSrc/src/main/kotlin/Dependencies.kt:108]: Done

**Review by jingtang10 — APPROVED** [review]

**Pull request merged**